pkg/http/server.go

```diff
@@ -127,23 +127,76 @@ func Run(cfg config.MutualPeersConfig) {
 	log.Info("Server Exited Properly")
 }
 
-func GenerateHashMetrics(cfg config.MutualPeersConfig, err error) bool {
-	// Get the genesisHash
-	// check if the config has the consensusNode field defined
+// BackgroundGenerateHashMetric check if we have defined the consensus in the config, if so, it creates a goroutine
+// to generate the metric
+func BackgroundGenerateHashMetric(cfg config.MutualPeersConfig) {
+	// Check if the config has the consensusNode field defined to generate the metric from the Genesis Hash data.
+	log.Info("BackgroundGenerateHashMetric...")
+
 	if cfg.MutualPeers[0].ConsensusNode != "" {
-		blockHash, earliestBlockTime := nodes.GenesisHash(cfg)
-		err = metrics.WithMetricsBlockHeight(
-			blockHash,
-			earliestBlockTime,
-			cfg.MutualPeers[0].ConsensusNode,
-			os.Getenv("POD_NAMESPACE"),
-		)
-		if err != nil {
-			log.Errorf("Error registering metric block_height_1: %v", err)
-			return true
+		log.Info("Initializing goroutine to generate the metric: hash ")
+		// Initialise the goroutine to generate the metric in the background, only if we specify the node in the config.
+		go func() {
+			log.Info("Consensus node defined to get the first block")
+
+			// Watch for events on the watcher channel
+			done := make(chan error)
+			go WatchHashMetric(cfg, done)
+
+			// Handle errors from WatchHashMetric
+			for {
+				select {
+				case err := <-done:
+					if err != nil {
+						log.Error("Error in WatchHashMetric: ", err)
+						// Handle the error as needed
+					}
+				}
+			}
+		}()
+	}
+}
+
+// WatchHashMetric watches for changes to generate hash metrics in the specified interval
+func WatchHashMetric(cfg config.MutualPeersConfig, done chan<- error) {
+	for {
+		err := GenerateHashMetrics(cfg)
+		// check if err is nil, if so, that means that Torch was able to generate the metric.
+		if err == nil {
+			log.Info("Metric generated for the first block...")
+			// The metric was successfully generated, stop the retries.
+			done <- nil
+			return
 		}
+
+		// Wait for the retry interval before the next execution
+		time.Sleep(retryInterval)
 	}
```

Comment:
you need to be more precise with handling not expected behaviour here. I.e. we need to retry if we fail, hence it makes sense to put into the else. In addition, do we plan to watch until forever? If no, it's much better to set a clear deadline to this process (can be like 4-5 min) where you pass the context to a watcher function and then exit the for loop gracefully.

Moreover, I highly recommend to go with select here, if you opt for the option I provided above